tests/php/Security/MemberTest.php

```diff
@@ -1618,4 +1621,111 @@ public function testChangePasswordToBlankIsValidated()
         $result = $member->changePassword('');
         $this->assertFalse($result->isValid());
     }
+
+    public function testMapInCMSGroupsNoLeftAndMain()
+    {
+        if (class_exists(LeftAndMain::class)) {
+            $this->markTestSkipped('LeftAndMain must not exist for this test.');
+        }
+        $result = Member::mapInCMSGroups();
+        $this->assertInstanceOf(Map::class, $result);
+
+        $this->assertEmpty($result, 'Without LeftAndMain, no groups are CMS groups.');
+    }
+
+    /**
+     * @dataProvider mapInCMSGroupsProvider
+     */
+    public function testMapInCMSGroups(array $groupFixtures, array $groupCodes, array $expectedValue)
+    {
+        if (!empty($groupFixtures) && !empty($groupCodes)) {
+            throw new InvalidArgumentException('Data provider is misconfigured for this test.');
+        }
+
+        if (!class_exists(LeftAndMain::class)) {
+            $this->markTestSkipped('LeftAndMain must exist for this test.');
+        }
+
+        $groups = [];
+
+        // Convert fixture names to IDs
+        if (!empty($groupFixtures)) {
+            foreach ($groupFixtures as $index => $groupFixtureName) {
+                $groups[$index] = $this->objFromFixture(Group::class, $groupFixtureName)->ID;
+            }
+        }
+
+        // Convert codes to DataList
+        if (!empty($groupCodes)) {
+            $groups = Group::get()->filter(['Code' => $groupCodes]);
+        }
+
+        $result = Member::mapInCMSGroups($groups);
+        $this->assertInstanceOf(Map::class, $result);
+
+        $this->assertSame($expectedValue, $result->values());
+    }
+
+    public function mapInCMSGroupsProvider()
```

Comment:
```suggestion
    public function provideMapInCMSGroups()
```
Current convention is to change the leading word 'test' to 'provide' so it's easy to match the test and the provider, they have the same casing

Reply:
Done